protostar/commands/test/test_command.py

```diff
@@ -174,8 +247,16 @@ async def run(self, args: Namespace) -> TestingSummary:
             max_steps=args.max_steps,
             slowest_tests_to_report_count=args.report_slowest_tests,
             gas_estimation_enabled=args.estimate_gas,
+            json_format=json_format,
         )
         cache.write_failed_tests_to_cache(summary)
+        if json_format:
+            if self._messenger_factory is None:
+                raise ProtostarException(
+                    "Messanger factory has to be provided in order to format the output"
+                )
+            write = self._messenger_factory.json()
+            write(SuccessfulTestMessage(summary, test_collector_result))
```

Comment:
Why do you raise an error to the user? Why is the `MessangerOptional` optional?

Reply:
Not related anymore